src/base/bittorrent/session.h

```diff
@@ -50,10 +50,14 @@
 #include "sessionstatus.h"
 #include "torrentinfo.h"
 
-#if ((LIBTORRENT_VERSION_NUM >= 10206) && !defined(Q_OS_WIN))
+#if ((LIBTORRENT_VERSION_NUM >= 10206) && !defined(Q_OS_WIN)) || (LIBTORRENT_VERSION_NUM >= 10212)
 #define HAS_HTTPS_TRACKER_VALIDATION
 #endif
 
+#if ((10212 <= LIBTORRENT_VERSION_NUM) && (LIBTORRENT_VERSION_NUM < 20000)) || (LIBTORRENT_VERSION_NUM >= 20002)
```

Comment:
I would prefer to use the same notation as we use everywhere: ` LIBTORRENT_VERSION_NUM >= 10212`

Reply:
I also want to note that in the light of recent decisions, such conditionals are no longer acceptable. We will need to raise the minimum supported versions of libtorrent to 1.2.12 and 2.0.2, respectively, as soon as they are released.

Reply:
>I would prefer to use the same notation as we use everywhere: LIBTORRENT_VERSION_NUM >= 10212

OK, will change it.

>We will need to raise the minimum supported versions of libtorrent to 1.2.12 and 2.0.2, respectively, as soon as they are released.

But there is no ETA for the release date right? It doesn't make sense to wait for libtorrent and keep this PR on hold.

Reply:
> But there is no ETA for the release date right? It doesn't make sense to wait for libtorrent and keep this PR on hold.

:+1:

Reply:
>It doesn't make sense to wait for libtorrent and keep this PR on hold.

The question is, does it make sense to merge it until then?
libtorrent keeps the previous version until the next release, now it is still 1.2.11 (in RC_1_2 branch, and I think 2.0.1 in RC_2_0), so this PR does not add any benefit to the resulting code, because all this will be conditionally excluded in the end. No one can even test these functions directly from this PR without modifying the code.
So why not just delete all these conditionals, mark PR as "waiting upstream" and merge it only when support is really released?

Reply:
>so this PR does not add any benefit to the resulting code, because all this will be conditionally excluded in the end. 

I see what you mean, I'm considering to split this PR.

>So why not just delete all these conditionals, mark PR as "waiting upstream" and merge it only when support is really released?

IMO bumping the version requirements isn't a trivial step as it could depend on other uncontrollable factors, for example we don't want our CI to fail and part of our CI depends on vcpkg (libtorrent, Qt) and thus unpredictable ETA. In this sense, I don't think blocking feature-adding PRs is wise (due to CI can't catch up) and nor breaking CI is a sensible choice (due to merging new features). Including version conditionals first and drop it later (in a version bump PR) is just one easy way to circumvent the problem.

Reply:
>I'm considering to split this PR.

👍 
>Including version conditionals first and drop it later (in a version bump PR) is just one easy way to circumvent the problem ("feature adding" and "version bump" issues mixed together).

I agree.
However, it is highly undesirable for it to be released in this (unstable feature set) state. So we should either not release until the requirements are raised (and conditionals are removed), or we should not include this code in the release branch until the conditionals are removed.

Reply:
I've split part of this PR to #13895.

>or we should not include this code in the release branch until the conditionals are removed.

By "release branch" you mean "v4_3_x" branch? If so, then having "version conditionals" in master branch is acceptable?

Reply:
>By "release branch" you mean "v4_3_x" branch?

Yes.
>If so, then having "version conditionals" in master branch is acceptable?

I would say Yes. The only question is how much harder it will make adding subsequent commits in release branch.

Or just forget about it and make it easier? Just add similar code conditionally when we need/want it, and then raise libtorrent version requirement as soon as possible?

Reply:
>Just add similar code conditionally when we need/want it, and then raise libtorrent version requirement as soon as possible?

👍

Reply:
I disagree with having a different `#ifdef` policy in release vs. master branch. It makes branch management more complex. If CI is the only thing stopping us right now, and we want faster and more predictable libtorrent CI upgrades, we can just maintain our own custom overlay libtorrent port for vcpkg (for use with its `--overlay-ports` option), while we wait for the upgrade to be merged upstream. This also paves the way for testing with more than one version at the same time.